beacon_node/network/src/beacon_processor/work_reprocessing_queue.rs

```diff
@@ -770,6 +831,32 @@ impl<T: BeaconChainTypes> ReprocessQueue<T> {
                     }
                 }
             }
+            InboundEvent::ReadyBackfillSync(queued_backfill_batch) => {
+                let seconds_from_slot_start = slot_clock
+                    .seconds_from_current_slot_start()
+                    .map_or("unknown".to_string(), |duration| {
+                        duration.as_secs().to_string()
+                    });
+
+                debug!(
+                    log,
+                    "Sending scheduled backfill work to Beacon Processor at {} seconds from current slot start",
+                    seconds_from_slot_start
+                );
+
+                if self
+                    .ready_work_tx
+                    .try_send(ReadyWork::BackfillSync(queued_backfill_batch.clone()))
+                    .is_err()
+                {
+                    error!(
+                        log,
+                        "Failed to send scheduled backfill work. Sending it back to queue.";
+                    );
+                    self.queued_backfill_batches
+                        .insert(0, queued_backfill_batch);
```

Comment:
I was reviewing the logs of some of our nodes and saw this happening (the channel being full) and remembered that here you need to re create the sleep as well

Reply:
hmm interesting. Does it mean another backfill batch can arrive at `BeaconProcessor` before one gets processed? Shall we perhaps increase the queue size instead?

I'm just thinking If we recreate the sleep here, it also resets the time that next batch gets processed, so that could slow down the backfill 🤔 is it possible?

Reply:
no, what I mean is that if sending fails and you re add the batch, there is no sleep to be polled and unqueue  it again

Reply:
ah nvm I thought your sleep was an option. All good, this is fine

Reply:
ah ok cool - I'm also running this on Goerli, will keep checking the logs